File: Apps/HotelManagement/MobileApp/src/types/hotel.ts
export interface Chain {
  id: number;
  name: string;
  headquarters: string;
  foundedYear: number;
  totalHotels: number;
  website: string;
  logoUrl: string;
}

export interface Brand {
  id: number;
  name: string;
  chainName: string;
  segment: string;
  logoUrl: string;
  description: string;
  minStarRating: number;
  maxStarRating: number;
}

export interface Hotel {
  id: number;
  name: string;
  brandId: number;
  brandName?: string;
  chainName?: string;
  city: string;
  country: string;
  address: string;
  starRating: number;
  phone: string;
  email: string;
  website: string;
  description: string;
  totalRooms: number;
  status: string;
  amenities: string[];
  images: string[];
  location: {
    latitude: number;
    longitude: number;
  };
}

export interface HotelStats {
  totalHotels: number;
  totalBrands: number;
  totalChains: number;
  hotelsByCountry: { country: string; count: number }[];
  hotelsByStarRating: { starRating: number; count: number }[];
}

File: Apps/HotelManagement/Backend/HotelManagement.API/Models/Hotel.cs
namespace HotelManagement.API.Models
{
    public class Hotel
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public int BrandId { get; set; }
        public string City { get; set; }
        public string Country { get; set; }
        public string Address { get; set; }
        public int StarRating { get; set; }
        public string Phone { get; set; }
        public string Email { get; set; }
        public string Website { get; set; }
        public string Description { get; set; }
        public int TotalRooms { get; set; }
        public DateTime OpeningDate { get; set; }
        public string Status { get; set; } // Active, Maintenance, Closed
        public List<string> Amenities { get; set; }
        public List<string> Images { get; set; }
        public GeoLocation Location { get; set; }
    }

    public class GeoLocation
    {
        public double Latitude { get; set; }
        public double Longitude { get; set; }
    }
}